packages/flutter_tools/lib/src/dart/pub.dart

```diff
@@ -73,15 +81,29 @@ typedef String MessageFilter(String message);
 Future<Null> pub(List<String> arguments, {
   String directory,
   MessageFilter filter,
-  String failureMessage: 'pub failed'
+  String failureMessage: 'pub failed',
+  @required bool retry,
 }) async {
   final List<String> command = <String>[ sdkBinaryName('pub') ]..addAll(arguments);
-  final int code = await runCommandAndStreamOutput(
-    command,
-    workingDirectory: directory,
-    mapFunction: filter,
-    environment: <String, String>{ 'FLUTTER_ROOT': Cache.flutterRoot, _pubEnvironmentKey: _getPubEnvironmentValue() }
-  );
+  int attempts = 0;
+  int duration = 1;
+  int code;
+  while (true) {
+    attempts += 1;
+    code = await runCommandAndStreamOutput(
+      command,
+      workingDirectory: directory,
+      mapFunction: filter,
+      environment: <String, String>{ 'FLUTTER_ROOT': Cache.flutterRoot, _pubEnvironmentKey: _getPubEnvironmentValue() }
+    );
+    if (code != 69)
```

Comment:
It would be help to compare code with a constant, like WHAT_DOES_69_SIGNIFY_FOR_THOSE_NOT_IN_THE_KNOW

Reply:
Agreed. I've no idea what 69 signifies. :-(

Reply:
I can leave a comment though.

Reply:
Per https://github.com/dart-lang/pub/blob/master/lib/src/exit_codes.dart#L31 it means `A service is unavailable`